sdk/core/Azure.Core/src/RequestContent.cs

```diff
@@ -136,13 +145,19 @@ public static RequestContent Create(object serializable, JsonPropertyNames prope
         /// </summary>
         /// <param name="stream">The stream to write to.</param>
         /// <param name="cancellation">To cancellation token to use.</param>
+#if NET6_0_OR_GREATER
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode("Inherited methods require unreferenced code.")]
```

Comment:
Is it true for all implementations? If not, then a better design is to leave the method without annotation and make the derived type trim-incompatible. Especially if you put the `RequiresUnreferencedCode` onto the entire type it will:
- make all .ctors warn on call
- make all static methods warn on call
- suppress all warnings in all method bodies on the type